### PR TITLE
Add books , person and rentals association

### PR DESCRIPTION
**Pull request by PedroH-Neves**
Here in this pull request I: 

- I created some new classes like book, classroom and rentals
- I made the association with person, book and rentals

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label
+
+  def intializate(label)
+    @label = label
+    @classroom = []
+  end
```
Comment (Owusu-Desmond) [inline classroom.rb]: - [ ] Please in your Classroom class, there are typos in the initialize method. This method should be spelled as `initialize`, not `intializate`.

File: classroom.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Classroom
```
Comment (Owusu-Desmond) [inline classroom.rb:1]: - [ ] Please ensure that you add `attr_reader :students` to provide a getter method for accessing the students in the classroom. :pray:

File: classroom.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  attr_accessor :label
+
+  def intializate(label)
+    @label = label
+    @classroom = []
+  end
+
+  def add_student(student)
+    @classroom.push(student)
```
Comment (Owusu-Desmond) [inline classroom.rb]: - [ ] I suggest that you use the `<<` operator to add students to the classroom, which is the conventional way in Ruby :blush:

File: student.rb [hunk 4]
```diff
@@ -6,6 +6,11 @@ def intializate(classroom, age, name = 'Unknown', parent_permission: true)
     @classroom = classroom
   end
```
Comment (Owusu-Desmond) [inline student.rb:9]: - [ ] Please in your Student class, there are typos in the initialize method. This method should be spelled as `initialize`, not `intializate`.
- [ ] I suggest that you add `attr_reader :classroom` to provide a getter method for accessing the a classroom a student belong to 🙏

File: student.rb [hunk 5]
```diff
@@ -6,6 +6,11 @@ def intializate(classroom, age, name = 'Unknown', parent_permission: true)
...
 
+  def classroom_assign(classroom_assign)
```
Comment (Owusu-Desmond) [inline student.rb]: - [ ] I think you should replaced `classroom_assign` with `classroom=` to set the classroom for a student, aligning with the project requirements. :relaxed: This below method definition follows the convention for setter methods in Ruby, where you have a method with a name ending in `=` and a single parameter representing the value you want to set.
  
  ```ruby
  def classroom=(classroom)
        @classroom = classroom
        classroom.students.push(self) unless classroom.students.include?(self)
    end
  ```

File: book.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+class Book
+  attr_reader :rentals
+  attr_accessor :title, :author
+
+  def intializate(title, author)
+    @title = title
```
Comment (Owusu-Desmond) [inline book.rb]: - [ ] Kindly fixed typo in `initialize` to ensure that the constructor method is correctly defined.  this method should be spelled `initialize` or `intializate`

File: rental.rb [hunk 7]
```diff
@@ -0,0 +1,14 @@
+require './book'
+require './person'
+
+class Rental
+  attr_accessor :date, :book, :person
+
+  def intializate(date, book, person)
```
Comment (Owusu-Desmond) [inline rental.rb]: - [ ] Kindly fixed typo in `initialize` to ensure that the constructor method is correctly defined.  this method should be spelled `initialize` or `intializate`

File: book.rb [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals << rental
+    rental.book = self
+  end
```
Comment (Owusu-Desmond) [inline book.rb]: - [ ] Please make sure that `this method for adding rentals takes in the person and the date and the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. :pray: 
  
  ```ruby
  def add_rental(person, date)
     Rental.new(date, self, person)
   end
  ``` 


    
```

File: person.rb [hunk 9]
```diff
@@ -6,10 +6,16 @@ def intializate(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals << rental
+    rental.person = self
   end
```
Comment (Owusu-Desmond) [inline person.rb]: - [ ] Please make sure that this method for adding rentals takes in the book and the date and the person itself should be sent as a parameter to create the new rental achieving the 'has-many' association 🙏

  ```ruby
    def add_rental(book, date)
      Rental.new(date, book, self)
    end
  ```

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @PedroH-Neves , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

- [ ] **NOTE** :warning:  Please next time make sure that the branch should have the name of a feature or action (e.g. `created_sidebar` or `added_comments_section`, `created_form` or `created_form_section`). But for now, as you can see you have your branch `feature` which does not match what you worked on in this branch :smile:  you have to specify feature as the branch name.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
..

File: student.rb [hunk 10]
```diff
@@ -1,11 +1,16 @@
 require './person'
 
 class Studentes < Person
-  def intializate(classroom, age, name = 'Unknown', parent_permission: true)
+  def initialize(classroom, age, name = 'Unknown', parent_permission: true)
```
Comment (Owusu-Desmond) [inline student.rb]: - [ ] As suggested already, kindly add `attr_reader :classroom` to provide a getter method for accessing a classroom a student belong to.

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @PedroH-Neves , :wave: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some changes you still need to work on before you move on to the next project but you are almost there.



 ## Highlights :+1: 
:heavy_check_mark: Good commit history.
:heavy_check_mark: No linter errors.
:heavy_check_mark: Your project is well documented


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AbayomiOlaoye — APPROVED** [review]
## STATUS: APPROVED :green_circle:  

Hi @PedroH-Neves   :wave: :wave:, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

applause

## Highlights
- Cleanly written code :white_check_mark: 
- App runs well and functions as expected ✅ 
- Nicely implemented project requirement ✅ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**